pkg/terraform/store.go

```diff
@@ -121,6 +133,13 @@ func (ws *WorkspaceStore) Workspace(ctx context.Context, c resource.SecretClient
 		return nil, errors.Wrap(err, "cannot write main tf file")
 	}
 	l := ws.logger.WithValues("workspace", dir)
+	reattachConfig, err := ws.providerRunner.StartSharedServer()
+	if err != nil {
+		return nil, err
+	}
+	if err := os.Setenv(envReattachConfig, reattachConfig); err != nil {
```

Comment:
There will be concurrent calls to this `Workspace` function, so setting an OS-wide env var may not work well here. If it's a required env var for every TF CLI invocation, maybe we can set it as env var to `*Workspace` object retrieved from the store map?

Reply:
Hmm I thought it was process-safe, i.e. process setting its own execution, hence the forked tf calls would use the set of env vars at the moment they were forked.